gprofiler/main.py

```diff
@@ -817,6 +889,11 @@ def main() -> None:
         if client is not None and remote_logs_handler is not None:
             remote_logs_handler.init_api_client(client)
 
+        if hasattr(args, "func"):
```

Comment:
This runs after `verify_preconditions()`, so we still require root & init PID NS. Let's ensure these are not requried, I think best would be to refrain from calling `verify_preconditions()` if we're in `upload-file` mode.